### DISCUSSION TITLE
Cloning returning permission issues

### DISCUSSION THREAD

**Discussion by mixart** [ProjectOpenSea/seadrop#38] [Q&A] [answered]
When I'm trying to clone the repro via:
`git clone --recurse-submodules https://github.com/ProjectOpenSea/seadrop && cd seadrop`

I'm getting a bunch of errors, tried to look for any dependency issues, but not getting past this:

```
Cloning into '/Users/mickgow/seadrop/lib/ERC721A-Upgradeable'...
The authenticity of host 'github.com (192.30.255.112)' can't be established.
ED25519 key fingerprint is SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'github.com' (ED25519) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:chiru-labs/ERC721A-Upgradeable.git' into submodule path '/Users/mickgow/seadrop/lib/ERC721A-Upgradeable' failed
Failed to clone 'lib/ERC721A-Upgradeable'. Retry scheduled

```
... etc.

```
Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:OpenZeppelin/openzeppelin-contracts-upgradeable.git' into submodule path '/Users/mickgow/seadrop/lib/openzeppelin-contracts-upgradeable' failed

```

Any suggestions? I've had no probs closing the seaport repro, just this seadrop repro.

**Comment by SteversIO** [comment, marked as answer]
@mixart you need to set up your SSH key configuration properly in github and locally on your machine.

[https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey](https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey)

If you already have an ssh key setup then chances are it wasn't added to your ssh-agent and when you try to perform a clone with the git@github.com user it fails.

Output from this command will show you which keys were attempted.
```
ssh -vT git@github.com
```

You'll need to add your specific key using:
```
ssh-add /path/to/key
```
